jsm.go

```diff
@@ -340,18 +340,59 @@ func (c *ConsumerLister) Next() bool {
 }
 
 // Page returns the current ConsumerInfo page.
-func (c *ConsumerLister) Page() []*ConsumerInfo {
+func (c *consumerLister) Page() []*ConsumerInfo {
 	return c.page
 }
 
 // Err returns any errors found while fetching pages.
-func (c *ConsumerLister) Err() error {
+func (c *consumerLister) Err() error {
 	return c.err
 }
 
-// NewConsumerLister is used to return pages of ConsumerInfo objects.
-func (js *js) NewConsumerLister(stream string) *ConsumerLister {
-	return &ConsumerLister{stream: stream, js: js}
+// ConsumersInfo is used to retrieve a list of ConsumerInfo objects.
+func (jsc *js) ConsumersInfo(stream string, opts ...JSOpt) <-chan *ConsumerInfo {
+	var o js
+	for _, opt := range opts {
+		if err := opt.configureJSContext(&o); err != nil {
+			return nil
+		}
+	}
+	// Check for option collisions. Right now just timeout and context.
+	if o.ctx != nil && o.wait != 0 {
+		return nil
+	}
+	if o.wait == 0 && o.ctx == nil {
+		o.wait = jsc.wait
+	}
+	if o.ctx == nil && o.wait > 0 {
+		o.ctx, o.ctxCancel = context.WithTimeout(context.Background(), o.wait)
```

Comment:
I think we do not need `o.ctxCancel` as part of the opts state since we can use the closure when creating the goroutine below.